### PR TITLE
side panel resize

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -157,22 +156,25 @@ class App {
 
     // center panel, resizable
     self._view.sidepanel = yo`
-      <div id="side-panel" class=${css.sidepanel}>
+      <div id="side-panel" style="min-width: 320px;" class=${css.sidepanel}>
```
Comment (GrandSchtroumpf) [inline src/app.js]: Set the min-width in CSS. Don't set style inline except if you're modifing it with Javascript.

File: src/lib/panels-resize.js [hunk 2]
```diff
@@ -21,28 +20,25 @@ const css = csjs`
...
 export default class PanelsResize {
-  constructor (idpanel1, idpanel2, opt) {
-    var panel1 = document.querySelector(idpanel1)
-    var panel2 = document.querySelector(idpanel2)
-    this.panel1 = panel1
-    this.panel2 = panel2
-    this.opt = opt
+  constructor (panel) {
+    this.panel = panel
+    let self = this
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: remove self

File: src/lib/panels-resize.js [hunk 3]
```diff
@@ -21,28 +20,25 @@ const css = csjs`
...
+    const string = panel.style.minWidth
+    this.minWidth = string.length > 2 ? parseInt(string.substring(0, string.length - 2)) : 0
+    window.addEventListener('resize', function (event) {
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: Use arrow function to have acces to `this`: 
`window.addEventListener('resize', (event) => this.setPosition(event))`

File: src/lib/panels-resize.js [hunk 4]
```diff
@@ -21,28 +20,25 @@ const css = csjs`
...
+      self.setPosition(event)
+    })
+  }
 
-    var ghostbar = yo`<div class=${css.ghostbar}></div>`
+  render () {
+    this.ghostbar = yo`<div class=${css.ghostbar}></div>`
 
     let mousedown = (event) => {
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: use const

File: src/lib/panels-resize.js [hunk 5]
```diff
@@ -51,58 +47,47 @@ export default class PanelsResize {
 
     let cancelGhostbar = (event) => {
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: use const

File: src/lib/panels-resize.js [hunk 6]
```diff
@@ -51,58 +47,47 @@ export default class PanelsResize {
...
-      if (p.panel1Width <= opt.minWidth || p.panel2Width <= opt.minWidth) return
-      ghostbar.style.left = event.x + 'px'
-    }
-
-    let setPosition = (event) => {
-      let p = processPositions(event)
-      panel1.style.width = p.panel1Width + 'px'
-      panel2.style.left = p.panel2left + 'px'
-      panel2.style.width = p.panel2Width + 'px'
+    let moveGhostbar = (event) => {
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: use const

File: src/lib/panels-resize.js [hunk 7]
```diff
@@ -51,58 +47,47 @@ export default class PanelsResize {
...
 
-    var dragbar = yo`<div onmousedown=${mousedown} class=${css.dragbar}></div>`
-    panel1.appendChild(dragbar)
+  calculatePanelWidth (event) {
+    return event.x - this.panel.offsetLeft
+  }
 
-    setPosition(opt)
+  setPosition (event) {
+    let panelWidth = this.calculatePanelWidth(event)
```
Comment (GrandSchtroumpf) [inline src/lib/panels-resize.js]: If you don't reasign this value, use const

File: test-browser/commands/checkTerminalFilter.js [hunk 8]
```diff
@@ -17,10 +17,10 @@ function checkFilter (browser, filter, test, done) {
     done()
     return
   }
-  var filterClass = '#editor-container div[class^="search"] input[class^="filter"]'
+  var filterClass = '#main-panel div[class^="search"] input[class^="filter"]'
```
Comment (GrandSchtroumpf) [inline test-browser/commands/checkTerminalFilter.js]: use const

File: src/app.js (hunk 1 above)
Comment (LianaHus) [reply src/app.js]: this is done in purpose, I had to do this so I can use in Panel-resize.

**Pull request merged**